### PR TITLE
Add pytest for WFC3/IR SVM processing

### PR DESCRIPTION
**Pull request by stsci-hack**
This adds some pytests for the SVM processing of WFC3/IR data based on data from visit ib6807 and using the template for the tests developed under #1185 .

### CODE REVIEW

File: tests/hap/test_svm_wfc3ir.py [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+    test_svm_demo.py
+
+    This test file can be executed in the following manner:
+        $ pytest -s --basetemp=/internal/hladata/yourUniqueDirectoryHere test_svm.py >& test_svm.log &
+        $ tail -f test_svm.log
+      * Note: When running this test, the `--basetemp` directory should be set to a unique
+        existing directory to avoid deleting previous test output.
+      * The POLLER_FILE exists in the tests/hap directory.
+
+    *** The --basetemp does not seem to be working at this time!!!
```
Comment (mdlpstsci) [inline tests/hap/test_svm_wfc3ir.py:24]: Suggest to update the commentary to be specific to this test and remove the --basetemp warning which is obsolete due to the tmpdir addition.

File: tests/hap/test_svm_wfc3ir.py [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+        table_length = len(ascii.read(cat, format="ecsv"))
+        print("\ntest_svm_cat_sources. Number of sources in catalog {} is {}.".format(cat, table_length))
+        valid_tables[cat] = table_length > 0
+    bad_tables = [cat for cat in cat_files if not valid_tables[cat]]
+    assert len(bad_tables) == 0, f"Catalog file(s) {bad_tables} is/are unexpectedly empty"
+
+def test_svm_point_cats(gather_output_data):
+    # Check that the point catalogs have the expected number of sources
+    cat_files = [files for files in gather_output_data if files.lower().endswith("point-cat.ecsv")]
+    expected_sources = {'ir': 358}
```
Comment (mdlpstsci) [inline tests/hap/test_svm_wfc3ir.py:187]: Just a suggestion for this and the next test as this is developer preference.  Putting expected_sources here keeps the values with the test, or putting the variable at the top of the module keeps the  variable which likely to change as algorithms evolve in one handy location.

**Review by mdlpstsci — CHANGES_REQUESTED** [review]
Only modest suggestions though I do think updating the doc at the beginning of the file should reflect the current test.

File: tests/hap/test_svm_wfc3ir.py [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+
+    num_sources = {cat:len(ascii.read(cat, format="ecsv")) for cat in cat_files}
```
[the remark this answers is not in the document]
Comment (stsci-hack) [reply tests/hap/test_svm_wfc3ir.py:189]: Thanks!

File: tests/hap/test_svm_wfc3ir.py (hunk 2 above)
Comment (stsci-hack) [reply tests/hap/test_svm_wfc3ir.py]: They have been renamed (to unique values) and moved to the top of the module in 2893c91.  This does make it easier to maintain these tests.

File: tests/hap/test_svm_wfc3ir.py (hunk 1 above)
Comment (stsci-hack) [reply tests/hap/test_svm_wfc3ir.py]: Fixed in 2893c91.

**Comment by stsci-hack** [comment]
The checks are failing for expected reasons; specifically, that #1187 has not yet been merged while the test is using PASS/FAIL criteria (number of sources and expected WCSNAME, to be precise) derived from running under #1187.

**Review by mdlpstsci — APPROVED** [review]
Changes are approved.

**Pull request merged**